bundles/plugins/org.bonitasoft.studio.expression.editor/src/org/bonitasoft/studio/expression/editor/viewer/ExpressionViewer.java

```diff
@@ -1282,10 +1282,15 @@ public void setRefactorOperationToExecuteWhenUpdatingContent(final AbstractRefac
     private boolean executeOperation(final String newValue) {
         boolean hasBeenExecuted = false;
         if (operation != null) {
+            operation.clearItemToRefactor();
             final Object oldValue = getInput();
             if (oldValue instanceof Element && !newValue.equals(((Element) getInput()).getName())
                     || oldValue instanceof SearchIndex && !newValue.equals(((SearchIndex) getInput()).getName().getName())) {
-                operation.addItemToRefactor(newValue, getInput());
+                final SearchIndex searchIndex = (SearchIndex) getInput();
+                final SearchIndex newSearchIndex = EcoreUtil.copy(searchIndex);
+                newSearchIndex.getName().setContent(newValue);
+                newSearchIndex.getName().setName(newValue);
+                operation.addItemToRefactor(newSearchIndex, searchIndex);
```

Comment:
I stop the refactoring here but, it's definitively something to change. A simple ReferenceDiff should be enough. Moreover this has nothing to do in the ExpressionViewer
I'll make another PR later about that.